Add "Restore a new VM on another Harvester cluster" in backup (#93)
harvester/docs: docs/vm/backup-restore.md

1. Click `Create`.

The restore process can be viewed from the `Virtual Machines` page.

## Restore a new VM on another Harvester cluster

_Available as of v1.0.0_

> You can migrate backup to another Harvester cluster and create a new VM based on it.

Reviewer: ```suggestion
You can migrate a backup to another Harvester cluster and create a new VM based on it.
```

Author: Updated it. Thank you.

> You can migrate a backup to another Harvester cluster and create a new VM based on it.

Reviewer 2: Users can now restore a new VM on another cluster by leveraging the VM metadata & content backup feature.

Author: Updated it. Thank you.

> **Requirements:** The VM image name and content must be the same as the old cluster.

Reviewer: ```suggestion
**Requirements:** The VM image name and content must be the same as configured in the cluster where the backup was migrated from.
```

Author: Updated it. Thank you.

> ### Upload same VM images to a new cluster

Reviewer: ```suggestion
### Upload the same VM images to a new cluster
```

Author: Updated it. Thank you.

> **Requirements:** The VM image name and content must be the same as configured in the cluster where the backup was migrated from.

Reviewer 2: prefer to use 
```
!!! prerequisites "Prerequisites"
     You must manually configure the virtual machine images with the same name on the new cluster first, otherwise the virtual machines will be failed to recover.
```

Author: Updated it. Thank you.

> 1. Get VM image names in the old cluster.

Reviewer: ```suggestion
1. Get the VM image names in the current cluster.
```

Author: Updated it. Thank you.

```

> > kubectl get vmimage -o json | jq '.items[].metadata.name'

Reviewer 2: ```
$ kubectl get vmimages -A
NAMESPACE   NAME                               DISPLAY-NAME                              SIZE         AGE
default     image-79hdq                        focal-server-cloudimg-amd64.img           566886400    5h36m
default     image-l7924                        harvester-v1.0.0-rc2-amd64.iso            3964551168   137m
default     image-lvqxn                        opensuse-leap-15.3.x86_64-nocloud.qcow2   568524800    5h35m
```

Check the existing image name (normally starts with `image-`) and create the same one on the new cluster.

Author: Updated it. Thank you.

"image-dgf27"
```

> 2. Apply a VM image YAML with the same name and content in a new cluster.

Reviewer: ```suggestion
2. Apply a VM image YAML with the same name and content in the new cluster.
```

Author: Updated it. Thank you.

```
> cat <<EOF | kubectl apply -f -
apiVersion: harvesterhci.io/v1beta1
kind: VirtualMachineImage
metadata:
  name: image-dgf27
  namespace: default
spec:
  displayName: opensuse-leap-15.3.x86_64-nocloud.qcow2
  pvcName: ""
  pvcNamespace: ""
  sourceType: download
  url: http://download.opensuse.org/repositories/Cloud:/Images:/Leap_15.3/images/openSUSE-Leap-15.3.x86_64-NoCloud.qcow2
EOF
```

### Restore a new VM in a new cluster

> 1. Setup the same backup target in a new cluster.

Reviewer 2: Append more to the first point with `And the backup controller will automatically sync the backup metadata to the new cluster.`

Author: Updated it. Thank you.

2. Go to the `Backups` page.

> 3. Specify the new VM name and click `Create`.

Reviewer 2: 3. Select the synced VM backup metadata and choose to restore a new VM with a specified VM name.

Author: Updated it. Thank you.

> 4. A new VM will be restored using the backup volumes and metadata, and users can access it from the `Virtual Machines` page.

Reviewer: ```suggestion
4. A new VM will be restored using the backup volumes and metadata. You can access it from the `Virtual Machines` page.
```

Author: Updated it. Thank you.
